dcm2niix_gear/dcm2niix/arrange.py

```diff
@@ -118,6 +118,121 @@ def exit_if_archive_empty(archive_obj):
         os.sys.exit(1)
 
 
+def tally_files(dir_loc, print_out=False):
+    """
+    Function to profile directory. Note current exclusion of {'.DS_Store', '._*'}
+    from file_set and file_tree. # TODO move this exclusion?
+
+    Args:
+        dir_loc: directory to profile
+        [print_out]: print output
+    Returns:
+        file_set: set of file leaves
+        file_tree
+        file_name_path_dict
+    """
+
+    file_set = set()
+    file_name_path_dict = {}
+    file_tree = []
+    gen = os.walk(dir_loc, topdown=True)
+    next_gen = next(gen)
+    while next_gen != StopIteration:
+
+        # info on current folder and contents
+        (loc_i, dirs_i, files_i) = next_gen
+
+        # populate outputs
+        if len(files_i) > 0:
+
+            # set of files (short paths or leaves) found in directory
+            # TODO could probably just use keys() of dictionary formed below
+            file_set = file_set | set(
```

Comment:
I would make the exclusion a list of regex's, I could foresee excluding other things in the future, or even providing that as a configuration option, and then checking each exclusion wouldn't be efficient code.  You could do something like:

```
import regex as r
excludes = [r.compile(r'.DS_Store'),r.compile(r'\.\_*')]
files = {}
for file in file_ij:
    for exclude in excludes:
        if not exclude.matches(file_ij):
            files.add(file_ij)
```

Reply:
Good idea. Tried a lot of slight variations on this, what do you think of below?

    strs_to_exclude = [".DS_Store", "^._"]
    excludes = [re.compile(s) for s in strs_to_exclude]

         .
         .
         .
   
         file_set = file_set | set([
                file_ij for file_ij in files_i
                if all([not(exclude.search(file_ij))
                for exclude in excludes])])